src/components/app-shell.tsx

```diff
@@ -17,11 +18,15 @@ export default class AppShell extends React.Component<Props> {
       <div>
         <Switch>
           <Route path="/" exact component={App} />
-          <Route path="/edited" component={App} />
-          <Route path="/gist/:mode/:username/:id/:revision/:filename" component={App} />
-          <Route path="/examples/:mode/:example_name" component={App} />
-          <Route path="/custom/:mode" component={App} />
-          <Route path="/url/:mode/:compressed" component={App} />
+          <Route exact path="/edited" component={App} />
```

Comment:
What does exact change here?

Reply:
The exact param here disables the partial matching for a route and makes sure that it only returns the route if the path is an EXACT match to the current url. 

Actually i added this when i was rendering a 404 component on routes which are not defined , so i added exact param.

Reply:
what is partial matching? Can you give me an example?

Reply:
we are making use of Switch Component, which renders the first route that matches, Router doesn't look for a complete match .

```
<Route path="/home" component={home} />
<Route path="/home/room" component={room} />

```
when we go to http://app.com/home , no problem occurs and everything works fine .
But, if we went to http://app.com/home/room, it would again go through all of our defined routes and return the FIRST match it finds. React router does partial matching, so` /home `partially matches `/home/room`.

So, the exact param disables this partial matching

Reply:
I see. Thanks.